tests/background_cpu_test.py

```diff
@@ -67,85 +81,64 @@ def _CheckVMFromSpec(self, spec, num_working):
       self.assertEqual(vm0.RemoteCommand.call_count,
                        expected_remote_post_stop)
 
-
   def testWindowsVMCausesError(self):
     """ windows vm with background_cpu_threads raises exception """
-    with mock_flags.PatchFlags() as mocked_flags:
-      self.setupCommonFlags(mocked_flags)
-      mocked_flags['background_cpu_threads'].Parse(1)
-      mocked_flags['os_type'].Parse(os_types.WINDOWS)
-      config = configs.LoadConfig(ping_benchmark.BENCHMARK_CONFIG, {}, NAME)
-      spec = benchmark_spec.BenchmarkSpec(config, NAME, UID)
-      spec.ConstructVirtualMachines()
-      with self.assertRaisesRegexp(Exception, 'NotImplementedError'):
-        spec.Prepare()
-      with self.assertRaisesRegexp(Exception, 'NotImplementedError'):
-        spec.StartBackgroundWorkload()
-      with self.assertRaisesRegexp(Exception, 'NotImplementedError'):
-        spec.StopBackgroundWorkload()
+    self._mocked_flags['background_cpu_threads'].Parse(1)
```

Comment:
Parse sets the flag object's value and marks the flag as present (i.e. the user provided a value).
https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/tests/mock_flags.py#L47
MockFlags has become a minimized version of gflags.FlagValues that can behave similarly in all of PKB's uses of FLAGS after its initialization, with the difference that MockFlags does not require importing all flag definitions prior to use. This solves the problem where a test only imports a module where a flag value is read, but not the module where the flag is defined.